### PR TITLE
Add dashboard for WAL metrics

### PR DESCRIPTION
**Pull request by codesome**
/cc @tomwilkie @gouthamve

### CODE REVIEW

File: cortex-mixin/dashboards.libsonnet [hunk 1]
```diff
@@ -11,7 +11,7 @@
      else {}) +
 
     (if std.setMember('chunks', $._config.storage_engine)
-     then import 'dashboards/chunks.libsonnet'
+     then (import 'dashboards/chunks.libsonnet') + (import 'dashboards/wal.libsonnet')
```
Comment (gouthamve) [inline cortex-mixin/dashboards.libsonnet:14]: Can we put this in `chunks.libsonnet` itself as the WAL is part of chunks?

Comment (codesome) [reply cortex-mixin/dashboards.libsonnet]: Done

File: cortex-mixin/dashboards/chunks.libsonnet [hunk 2]
```diff
@@ -50,4 +50,43 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
+      .addPanel(
+        $.panel('Bytes per sample') +
+        $.queryPanel('avg(rate(cortex_ingester_wal_logged_bytes_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval]) / rate(cortex_ingester_ingested_samples_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval]))', 'bytes'),
+      )
+    )
+    .addRow(
+      $.row('Checkpoint')
+      .addPanel(
+        $.panel('Checkpoint creation/deletion / sec') +
+        $.queryPanel('rate(cortex_ingester_checkpoint_creations_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval])', '{{instance}}-creation') +
```
Comment (gouthamve) [inline cortex-mixin/dashboards/chunks.libsonnet:83]: I am not sure a per instance graph would be useful here

File: cortex-mixin/dashboards/chunks.libsonnet [hunk 3]
```diff
@@ -50,4 +50,43 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
+        $.queryPanel('rate(cortex_ingester_checkpoint_deletions_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval])', '{{instance}}-deletion'),
+      )
+      .addPanel(
+        $.panel('Checkpoint creation/deletion failed / sec') +
+        $.queryPanel('rate(cortex_ingester_checkpoint_creations_failed_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval])', '{{instance}}-creation') +
+        $.queryPanel('rate(cortex_ingester_checkpoint_deletions_failed_total{cluster=~"$cluster", job=~"($namespace)/ingester"}[$__interval])', '{{instance}}-deletion'),
+      )
+    ),
```
Comment (gouthamve) [inline cortex-mixin/dashboards/chunks.libsonnet:91]: Can we add a panel that exports the min(free_space) on the mounted disk, etc.

File: cortex-mixin/dashboards/chunks.libsonnet (hunk 2 above)
Comment (codesome) [reply cortex-mixin/dashboards/chunks.libsonnet:83]: It is handy to click on a single instance and see when the checkpointing happened for it. If we have a sum of them, it doesn't seem to be that useful, or can you think of a use-case?

**Pull request merged**